test/VocabularyGeneratorTest.cpp

```diff
@@ -0,0 +1,175 @@
+#include <gtest/gtest.h>
+#include <string>
+#include <fstream>
+#include <cstdlib>
+#include <iostream>
+#include <ctime>
+
+#include "../src/index/ConstantsIndexCreation.h"
+#include "../src/index/VocabularyGenerator.h"
+#include "../src/global/Constants.h"
+
+// Test fixture that sets up the binary files vor partial vocabulary and
+// everything else connected with vocabulary merging.
+class MergeVocabularyTest : public ::testing::Test {
+protected:
+
+  // path of the 2 partial Vocabularies that are used by mergeVocabulary
+  std::string _path0;
+  std::string _path1;
+  // path of the 2 partial vocabularies that are the expected output of
+  // mergeVocabulary
+  std::string _pathExp0;
+  std::string _pathExp1;
+  // the base directory for our test
+  std::string _basePath;
+  // path to expected vocabulary text file
+  std::string _pathVocabExp;
+  // path to expected external vocabulary text file
+  std::string _pathExternalVocabExp;
+
+  // Constructor. TODO: Better write Setup method because of complex logic which
+  // may throw?
+  MergeVocabularyTest() {
+    // name of random subdirectory
+    std::srand(std::time(nullptr));
+    _basePath = std::string("qleverVocTest-") + std::to_string(std::rand());
+    // those names are required by mergeVocabulary
+    _path0 = std::string(PARTIAL_VOCAB_FILE_NAME + std::to_string(0));
+    _path1 =  std::string(PARTIAL_VOCAB_FILE_NAME + std::to_string(1));
+    // those names can be random
+    _pathExp0 = std::string(".partialVocabExp0");
```

Comment:
I thought we moved to `snake_case_names`?

Reply:
I think for filenammes it was names-with-dashes